### PR TITLE
Stats: Swapping Search and Countries components

### PR DESCRIPTION
**Pull request by grzegorz-cp**
#### Proposed Changes

* using a feature flag and replacing `Search` and `Countries` StatsModules with the recently created component with horizontal bars
* extending horizontal bar and stats card components to accept a hero item placed before the component's header (e.g. world map), allowing for items that don't have any click interaction
* adding a component to render country flags

#### Testing Instructions

* Navigate to the main Stats page either on a local instance or the test URL from the comment below
* append `?flags=stats/new-stats-module-component` to the URL
* verify that `Search` component has horizontal bars
* verify that `Countries` component has horizontal bars

#### Pre-merge Checklist

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

Related to #69044

### CODE REVIEW

**Review by jsnmoon — CHANGES_REQUESTED** [review]
Changes look reasonable, but I'm running into this bug with the Search module:

image

File: client/my-sites/stats/stats-list/stats-list-card.jsx [hunk 1]
```diff
@@ -29,15 +31,20 @@ const StatsListCard = ( {
...
 		}
 	};
 
-	const barMaxValue = data?.[ 0 ]?.value || 0;
+	let barMaxValue = 0;
+
+	// Search doesn't have items sorted by value when there are 'Unknown search terms' present.
+	data?.forEach( ( item ) => {
+		barMaxValue = item?.value > barMaxValue ? item?.value : barMaxValue;
+	} );
```
Comment (jsnmoon) [inline client/my-sites/stats/stats-list/stats-list-card.jsx]:
```suggestion
    // Search doesn't have items sorted by value when there are 'Unknown search terms' present.
	const barMaxValue = Math.max( ...data?.map( item => item.value ).filter( Number.isFinite ) );
```

A little code golf for fun 🙂 
(Warning: Hasn't been checked for edge cases!)

**Review by jsnmoon — APPROVED** [review]
Looks good! Ship whenever you're ready.

File: client/my-sites/stats/stats-list/stats-list-card.jsx (hunk 1 above)
Comment (grzegorz-cp) [reply client/my-sites/stats/stats-list/stats-list-card.jsx]: Oh, I like it! Thanks, I'll commit and test.

**Pull request merged**